### PR TITLE
update for GHC-7.10, -Wall

### PR DESCRIPTION
**Pull request by bergey**
Most of Foldable & Traversable are now in Prelude, so importing these
libraries is redundant.  I've just disabled those warnings.

### CODE REVIEW

File: src/Diagrams/Core/Names.hs [hunk 1]
```diff
@@ -1,11 +1,12 @@
+{-# LANGUAGE CPP                        #-}
```
Comment (byorgey) [inline src/Diagrams/Core/Names.hs]: Why did you add `CPP` here?

File: src/Diagrams/Core/Names.hs [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 {-# LANGUAGE DeriveDataTypeable         #-}
 {-# LANGUAGE FlexibleInstances          #-}
 {-# LANGUAGE GADTs                      #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE MultiParamTypeClasses      #-}
-{-# LANGUAGE OverlappingInstances       #-}
```
Comment (byorgey) [inline src/Diagrams/Core/Names.hs]: Huh, was this module not actually making use of `OverlappingInstances`?

File: src/Diagrams/Core/Names.hs (hunk 1 above)
Comment (bergey) [inline src/Diagrams/Core/Names.hs]: I'm pretty sure I added CPP to selectively turn off OverlappingInstances, and then found out that the module didn't need the latter.

**Comment by bergey** [comment]
I spoke too soon about merging this.  In `Names.hs`, I was tripped up by the fact that overlapping instances compile fine, and errors only appear where they are used.  Now `diagrams-contrib` fails to build: https://travis-ci.org/diagrams/diagrams-contrib/jobs/52657142

I think the overlapping instances are 

``` haskell
instance IsName Char
instance IsName String
instance IsName a => IsName [a]
```

Can we just drop the `String` instance?  If not, I'll fix it up with CPP.

**Comment by cchalmers** [comment]
Yeah, just remove the `String` instance. Seems like it was a mistake to be there in the first place.

**Comment by byorgey** [comment]
Indeed, I don't know of any good reason to have a separate (overlapping)
`String` instance.
